lib/Semantics/resolve-names.cpp

```diff
@@ -985,11 +985,18 @@ class ConstructVisitor : public virtual DeclarationVisitor {
   void Post(const parser::EndAssociateStmt &);
   void Post(const parser::Association &);
   void Post(const parser::SelectTypeStmt &);
+
+  void Post(const parser::SelectRankStmt &);
   bool Pre(const parser::SelectTypeConstruct &);
   void Post(const parser::SelectTypeConstruct &);
   bool Pre(const parser::SelectTypeConstruct::TypeCase &);
   void Post(const parser::SelectTypeConstruct::TypeCase &);
+  // gives blank scopes
```

Comment:
I do not understand this comment; please make it more clear.

Reply:
How about this?
// Creates Block scopes with no details nor symbol name.
I meant that printing the symbols using `-fdebug-dump-symbols `with this prints something like
```
Block scope:
Block scope:
Block scope:
Block scope:
```
I shall remove if that is misleading..